src/ccbr_tools/jobby.py

```diff
@@ -1,804 +1,272 @@
-def get_toolkit(tool_list):
-    """
-    Finds the best suited tool from a list of possible choices. Assumes tool list is already
-    ordered from the best to worst choice. The first tool found in a user's $PATH is returned.
-
-    Args:
-        tool_list (list[str]): List of ordered tools to find.
-
-    Returns:
-        str: First tool found in tool_list.
-
-    Raises:
-        SystemExit: If no tools are found in the user's $PATH.
-    """
-    best_choice = None
-    for exe in tool_list:
-        if which(exe):
-            best_choice = exe
-            break
-
-    # Did not find any tools
-    # to potentially use
-    if not best_choice:
-        err("Error: Did not find any tools to get job information!")
-        fatal(
-            "Expected one of the following tools to be in $PATH:"
-            "\t{0}".format(tool_list)
-        )
-
-    return best_choice
-
-
-def add_missing(linelist, insertion_dict):
-    """
-    Adds missing information to a list. This can be used
-    to add missing job information fields to the results of
-    job querying tool.
-
-    Args:
-        linelist (list[str]): List containing job information for each field of interest.
-        insertion_dict (dict[int, Union[str, list[str]]]): Dictionary used to insert missing information
-            to a given index, where the keys are indices of the `linelist` and the values are information
-            to add. The indices should be zero-based. Multiple consecutive values should be inserted at
-            once as a list.
-
-    Returns:
-        list[str]: The updated list with the missing information added.
-
-    Example:
-        add_missing([0,1,2,3,4], {3:['+','++'], 1:'-', 4:'@'})
-        >> [0, '-', 1, 2, '+', '++', 3, '@', 4]
-    """
-    # Get the order of indices
-    # add missing information
-    # starting from largest to
-    # smallest, if we insert
-    # missing values in this
-    # order we do not need to
-    # calculate the offset of
-    # new indices
-    tmp_list = linelist
-    indices = sorted(list(insertion_dict.keys()), reverse=True)
-    for i in indices:
-        # Check if multiple values
-        # need to be inserted at a
-        # given index
-        if isinstance(insertion_dict[i], list):
-            for v in reversed(insertion_dict[i]):
-                tmp_list.insert(i, v)
+try:
+    import yaml
+except ImportError:
+    yaml = None  # YAML is optional — we'll check later when needed
+
+# Columns we can extract using sacct
+COLUMNS = {
+    "JobID": "JobId",
+    "JobName": "JobName",
+    "State": "JobState",
+    "Elapsed": "RunTime",
+    "AllocNodes": "NumNodes",
+    "AllocCPUS": "NumCPUs",
+    "TotalCPU": "TotalCPUTime",
+    "ReqMem": "ReqMem",
+    "MaxRSS": "MaxMemUsed",
+    "ExitCode": "ExitCode",
+    "Timelimit": "Timelimit",
+    "NodeList": "NodeList",
+    "Start": "StartTime",
+    "End": "EndTime",
+    "Submit": "QueuedTime",
+    "WorkDir": "WorkDir",
+}
+
+
+def parse_time_to_seconds(t):
```

Comment:
it would be easier to understand this function if it had only one return statement at the end instead of several throughout